plugins/extraction/ocaml.ml

```diff
@@ -599,11 +599,30 @@ let pp_decl = function
         hov 2 (str "type " ++ ids ++ name ++ def)
     | Dterm (r, a, t) ->
         let def =
-          if is_custom r then str (" = " ^ find_custom r)
+          (* If it it is an foreign custom, set the def to ': type =  <quote>foreign_fun_name<quote> '.
+             TODO: I'm not sure, but could we use pp_native_string for quoting the custom name of r?
+          *)
+          if is_foreign_custom r then str (": ") ++ pp_type false [] t ++ str ( " = \"" ^ find_custom r ^ "\"")
+          (* Otherwise, check if it is a regular custom term. *)
+          else if is_custom r then str (" = " ^ find_custom r)
           else pp_function (empty_env ()) a
         in
         let name = pp_global_name Term r in
-        pp_val name t ++ hov 0 (str "let " ++ name ++ def ++ mt ())
+        (* If it is an foreign custom, begin the expression with 'external'/'foreign' instead of 'let' *)
+        let expr_begin = if (is_foreign_custom r) then str "external " else str "let " in
+        (* Make sure that a callback registration is synthesised, if specified for that term. *)
+        let callback_def =
+          try
+            let alias =
+              match (find_callback r) with
+                | Some s -> str s
+                | None   -> name (* No alias specified, use the qualid name. *)
+            in (* TODO: Is using let _ safe here? It does work in OCaml code.*)
+            cut2 () ++ hov 0 ((str "let _ = Callback.register \"") ++ alias ++ (str "\" ") ++ name)
```

Comment:
maybe should be 
```suggestion
            cut2 () ++ hov 0 ((str "let _ = Stdlib.Callback.register \"") ++ alias ++ (str "\" ") ++ name)
```
to reduce the chance of collision in case there's a Coq module named Callback around.

Reply:
True, you have a point here, thank you